src/internal/_toString.js

```diff
@@ -0,0 +1,37 @@
+var _indexOf = require('./_indexOf');
+var _map = require('./_map');
+var _quote = require('./_quote');
+var _toISOString = require('./_toISOString');
+var keys = require('../keys');
+
+
+module.exports = function _toString(x, seen) {
+    var recur = function recur(y) {
+        var xs = seen.concat([x]);
+        return _indexOf(xs, y) >= 0 ? '<Circular>' : _toString(y, xs);
+    };
+
+    if (x == null) {
+        return String(x);
+    } else if (x instanceof Boolean) {
+        return 'new Boolean(' + recur(x.valueOf()) + ')';
+    } else if (x instanceof Number) {
+        return 'new Number(' + recur(x.valueOf()) + ')';
+    } else if (x instanceof String) {
+        return 'new String(' + recur(x.valueOf()) + ')';
+    } else if (x.constructor === Array) {
+        return '[' + _map(recur, x).join(', ') + ']';
+    } else if (x.constructor === Date) {
+        return 'new Date(' + _quote(_toISOString(x)) + ')';
+    } else if (x.constructor === Number) {
+        return 1 / x === -Infinity ? '-0' : x.toString(10);
+    } else if (x.constructor === String) {
+        return _quote(x);
+    } else if (Object.prototype.toString.call(x) === '[object Arguments]') {
+        return '(function() { return arguments; }(' + _map(recur, x).join(', ') + '))';
+    } else if (x.toString !== Object.prototype.toString) {  // Boolean, Function, RegExp, user-defined types
+        return x.toString();
+    } else {
+        return '{' + _map(function(k) { return _quote(k) + ': ' + recur(x[k]); }, keys(x).sort()).join(', ') + '}';
+    }
+};
```

Comment:
Third try at this comment...

These conditionals have [cross-frame issues].  Should we care about this?

My gut feeling is that since it's straightforward enough to make this more robust, we probably should do so.  Others?

  [cf]: http://perfectionkills.com/instanceof-considered-harmful-or-how-to-write-a-robust-isarray/